io/serialport/25-serial.js

```diff
@@ -200,6 +222,45 @@ module.exports = function(RED) {
     }
     RED.nodes.registerType("serial request", SerialRequestNode);
 
+
+    // Serial Control Node
+    function PortSelectNode(n) {
+        const configProps = [
+            "serialport", "serialbaud", "databits", "parity", "stopbits",
+            "dtr", "rts", "cts", "dsr", "bin", "out"
+        ]
+        RED.nodes.createNode(this,n);
+        this.serialConfig = RED.nodes.getNode(n.serial);
+
+        if (!this.serialConfig) {
+            this.error(RED._("serial.errors.missing-conf"), {});
+            return;
+        }
+
+        this.serial = n.serial;
+        var node = this;
+        node.port = serialPool.get(this.serialConfig);
+        node.on("input",function(msg) {
+            if (configProps.some((p) =>{return msg.payload.hasOwnProperty(p)})) { 
+                msg.payload.enable = msg.payload.hasOwnProperty('enable') ? msg.payload.enable : true;
+                node.serialConfig.changePort(msg.payload);
+            }
+            if (msg.payload.hasOwnProperty("enable")) { 
+            // if any of config parameters or enable property is passed, do this control
+                node.serialConfig.enable = msg.payload.enable;
+                if (msg.payload.enable === true) {
+                    node.serialConfig.emit('start');
+                } else {
+                    serialPool.close(node.serialConfig.serialport,() => {
+                        RED.log.info("[serialconfig:"+node.serialConfig.id+"] " + RED._("serial.stopped",{port:node.serialConfig.serialport}));
+                    });
+                }
+            }
+            node.send({payload: node.serialConfig});
```

Comment:
This is a specific serialConfig object so appears to debug as "[object Object]"... plus it has all the node internal config in it... This needs to be **just** the parameters you can change/set so that it could be simply modified and sent back into the control node. (as it was in your first iteration)